platforms/chibios/timer.c

```diff
@@ -73,6 +73,12 @@ void timer_clear(void) {
     chSysUnlock();
 }
 
+void timer_set(uint32_t time_ms) {
+    ticks_offset = get_system_time_ticks();
+    last_ticks   = 0;
+    ms_offset    = time_ms;
+}
```

Comment:
This should be done while holding the ChibiOS system lock (one asynchronous thing that might interfere is `update_fn()`, which is used for 16-bit timers — it also calls `get_system_time_ticks()`).  However, maybe the resume code already holds the system lock at the time it needs to call `timer_set()`, in which case adding the locking here would actually break things.

Reply:
My assumption was if you're already setting up a bunch of a clocks again it's likely within chSysLock anyway. It is in my case, so I've purposely left it out.

Reply:
That chSysLock should be held when calling `timer_set()` is an important precondition. IMO, this deserves a doc comment in the .h file above the function declaration.

Reply:
After a little bit of discussion chSysLock and chSysUnlock have been added to timer_set. 

While this does mean it can't be called from within a system lock, it's unlikely that waiting to call it after an unlock will be of any detriment.